spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -193,6 +204,8 @@ private Expression filterExpression() {
 
   private StructType lazyType() {
     if (type == null) {
+      Preconditions.checkArgument(readTimestampWithoutZone || !hasTimestampWithoutZone(lazySchema()),
+              "Spark does not support timestamp without time zone fields");
```

Comment:
I would like this to be a bit more elaborate. Something like
"Cannot handle **timestamp without timezone** fields in Spark. Spark does not natively support this type but if you would like to handle all **timestamps** as **timestamp with timezone** set 'flag name' to true. This will not change the underlying values stored but will change their displayed values in Spark. For more information see ... `some website reference?`"

Reply:
I have been looking for a website to put as a reference in the error message and I'm having a hard time finding a specific site that talks about "Spark not supporting timestamp without timezone" fields. Could I maybe please get some pointers as to where I should look. I saw some relevant information on this site :

https://docs.databricks.com/spark/latest/dataframes-datasets/dates-timestamps.html

Reply:
https://docs.databricks.com/spark/latest/dataframes-datasets/dates-timestamps.html#ansi-sql-and-spark-sql-timestamps - That is probably enough info

Reply:
addressed it